app/questionnaire/router.py

```diff
@@ -0,0 +1,52 @@
+class Router:
+
+    def __init__(self, schema, routing_path, completeness, current_location=None):
+        self._schema = schema
+        self._routing_path = routing_path
+        self._completeness = completeness
+        self._current_location = current_location
+
+    def can_access_location(self):
+        if self._is_valid_location() and not self._get_first_incomplete_block_in_section() and not self._get_first_incomplete_block_in_survey():
+            return True
+
+        return False
+
+    def get_next_location(self):
+        if not self._is_valid_location():
+            next_location = self._get_expected_location()
+            return next_location
+
+        return self._get_first_incomplete_block_in_section() or self._get_first_incomplete_block_in_survey() or self._get_expected_location()
+
+    def _is_valid_location(self):
+        return self._current_location in self._routing_path
+
+    def _get_expected_location(self):
```

Comment:
Took me a while to work out what this method was responsible for. It's designed to work out what location to take a user to if there isn't a valid `current_location` available. I think the word "expected" is quite confusing here.

Perhaps consider calling it `_get_default_location` and then add a decent docstring to explain that it's essentially used for a) the location after login and b) as a fallback if somebody hits a URL they can't reach.

Reply:
Its used in the event that current location is invalid or current location is valid but not skipping.

Ill add doc strings and rename method 👍